retrieval/k8s/k8s_stats.go

```diff
@@ -1,82 +1,269 @@
 package k8s
 
 import (
-	"net/http"
+	"bufio"
+	"encoding/json"
+	"errors"
+	v1apps "k8s.io/api/apps/v1"
+	v1batch "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
 	"os"
+	"time"
+)
 
-	"github.com/cloudability/metrics-agent/retrieval/raw"
-	log "github.com/sirupsen/logrus"
+const (
+	KubernetesLastAppliedConfig = "kubectl.kubernetes.io/last-applied-configuration"
 )
 
-//GetK8sMetrics returns cloudabilty measurements retrieved from a given K8S Clientset
-func GetK8sMetrics(clusterHostURL string, clusterVersion float64, workDir *os.File, rawClient raw.Client) (err error) {
+func StartUpInformers(clientset kubernetes.Interface, clusterVersion float64,
+	resyncInterval int, stopCh chan struct{}) (map[string]*cache.SharedIndexInformer, error) {
+	factory := informers.NewSharedInformerFactory(clientset, time.Duration(resyncInterval)*time.Hour)
 
-	v1Sources, v1beta1Sources, v1AppSources := getk8sSourcePaths(clusterVersion)
+	// v1Sources
+	replicationControllerInformer := factory.Core().V1().ReplicationControllers().Informer()
```

Comment:
Minor: making `factory.Core().V1()` a variable would make this much more readable if that wont break anything

```suggestion
	replicationControllerInformer := factory.Core().V1().ReplicationControllers().Informer()
```

Reply:
I can do that, but we would need 3 variables one for each resource "group/type" that we collect on:
- factory.Core().V1()
- factory.Batch().V1()
- factory.Apss().V1()
Examples:
	replicasetsInformer := factory.Apps().V1().ReplicaSets().Informer()
	jobsInformer := factory.Batch().V1().Jobs().Informer()
	nodesInformer := factory.Core().V1().Nodes().Informer()
If you still think it is worth it, I am fine adding it.

Reply:
Nah all good.